library/PracticeManager.unit.test.js

```diff
@@ -0,0 +1,100 @@
+const PracticeManager = require('./PracticeManager')
+const DateMock = require('jest-date-mock')
+const moment = require('moment')
+
+// Some arbitrary userIds
+const prackUserId = 5
+const idleUserId = 11
+
+// Arbitrary fixed start time
+const startTime = new Date(2019, 2, 1, 0, 0, 0)
+const fiveSeconds = moment.duration(5, 'seconds')
+const anHour = moment.duration(1, 'hour')
+
+// Subject under test
+let pm
+beforeEach(() => {
+  pm = new PracticeManager()
+  DateMock.advanceTo(startTime)
+})
+
+test('can track whether pracking', () => {
+  pm.startPractice(prackUserId)
+  expect(pm.isPracticing(prackUserId)).toBe(true)
+  pm.stopPractice(prackUserId)
+  expect(pm.isPracticing(prackUserId)).toBe(false)
+})
+
+test('measures live practice time', () => {
+  pm.startPractice(prackUserId)
+  DateMock.advanceBy(fiveSeconds.asMilliseconds())
+
+  expect(pm.currentPracticeTime(prackUserId)).toBe(fiveSeconds.asSeconds())
+})
+
+test('ignores non-prackers', () => {
+  expect(pm.isPracticing(idleUserId)).toBe(false)
+  expect(pm.currentPracticeTime(idleUserId)).toBe(0)
+})
+
+test('starts with zero session time', () => {
+  expect(pm.sessionPracticeTime(prackUserId)).toBe(0)
```

Comment:
Again for my clarification - this is only true in the absence of persistence, yes?

Reply:
Well, depending on how we want the test to be set up, yes.